networkx/algorithms/assortativity/connectivity.py

```diff
@@ -17,7 +17,7 @@ def _avg_deg_conn(G, neighbors, source_degree, target_degree,
     dsum = defaultdict(float)
     dnorm = defaultdict(float)
     for n,k in source_degree(nodes).items():
-        nbrdeg = target_degree(neighbors(n))
+        nbrdeg = target_degree(list(neighbors(n)))
```

Comment:
#1607 is supposed to fix this in master. 
I think it was introduced in 2010 when we made changes for python 3.